packages/cli/src/global_telemetry.ts

```diff
@@ -84,20 +85,45 @@ export function initializeInstrumentation() {
 export function setupTelemetry(config: any, opts: any) {
   const now = new Date()
   const cmdStartTime = now.getTime()
-  return {
-    command: opts.Command.id,
-    os: config.platform,
-    version: config.version,
-    exitCode: 0,
-    exitState: 'successful',
-    cliRunDuration: 0,
-    commandRunDuration: cmdStartTime,
-    lifecycleHookCompletion: {
-      init: true,
-      prerun: true,
-      postrun: false,
-      command_not_found: false,
-    },
+  const isHelpOrVersionCmd = (opts.id === 'version' || opts.id === '--help')
+  const isRegularCmd = Boolean(opts.Command)
+
+  if (isHelpOrVersionCmd) {
```

Comment:
Todo: Create base object and add necessary properties instead of duplicating telemetry object